src/map/clif.c

```diff
@@ -12922,6 +12922,8 @@ static void clif_parse_UseSkillMap(int fd, struct map_session_data *sd)
 
 	pc->delinvincibletimer(sd);
 	skill->castend_map(sd,skill_id,map_name);
+
+	pc->itemskill_clear(sd);
```

Comment:
probably clear need call before invoking skill?

Reply:
Nope. `skill_castend_map()` calls `skill_check_condition_castend()` and `skill_consume_requirement()`, which both use itemskill flags. If you'd clear those flags before calling `skill_castend_map()`, those flags won't work.